svelte/src/components/tabs/tabs.svelte

```diff
@@ -1,49 +1,36 @@
 <script lang="ts">
   import type { Tab } from "$libs/types";
-  import { afterUpdate } from "svelte";
+  import { tabStore } from "$libs/stores";
 
-  let clazz = "";
+  const { activeTab, setActiveTab } = tabStore;
 
+  let clazz = "";
   export { clazz as class };
 
   import Button from "../button/button.svelte";
 
   export let tabs: Tab[] = [];
-  export let defaultTab: string;
-
-  let active: string;
-
-  let dirty = false;
 
-  afterUpdate(() => {
-    if (tabs.length && !active) {
-      if (!defaultTab) {
-        active = tabs[0].label;
-      } else if (!dirty) {
-        active = defaultTab;
-      }
-    }
-  });
+  $: activeTabLabel = $activeTab ?? tabs[0]?.label;
 </script>
 
 <section class="relative h-auto {clazz}">
   <menu class="flex gap-1">
     {#each tabs as tab}
-      <div class="border-gray text-white" class:border-b={tab.label === active}>
-        <Button
-          onClick={() => {
-            dirty = true;
-            active = tab.label;
-          }}
-        >
-          <span class:text-white={tab.label === active}>{tab.label}</span>
+      <div
+        class="border-gray text-white"
+        class:border-b={tab.label === activeTabLabel}
+        class:hidden={!!tab.hidden}
+      >
+        <Button onClick={() => setActiveTab(tab.label)}>
+          <span class:text-white={tab.label === activeTabLabel}>{tab.label}</span>
         </Button>
       </div>
     {/each}
   </menu>
 
   {#each tabs as tab}
-    {#if tab.label === active}
+    {#if tab.label === activeTabLabel}
```

Comment:
might help in not refreshing the iframe, just render all tabs and display:none

Reply:
sweet, i'll give that a try.